src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -117,14 +116,16 @@ public DocumentDatabase(string name, RavenConfiguration configuration, ServerSto
                 {
                     MasterKey = serverStore.GetSecretKey(ctx, Name);
 
-                    var databaseRecord = _serverStore.Cluster.ReadDatabase(ctx, Name);
-                    if (databaseRecord != null)
+                    using (var rawRecord = _serverStore.Cluster.ReadRawDatabaseRecord(ctx, Name))
                     {
-                        // can happen when we are in the process of restoring a database
-                        if (databaseRecord.Encrypted && MasterKey == null)
-                            throw new InvalidOperationException($"Attempt to create encrypted db {Name} without supplying the secret key");
-                        if (databaseRecord.Encrypted == false && MasterKey != null)
-                            throw new InvalidOperationException($"Attempt to create a non-encrypted db {Name}, but a secret key exists for this db.");
+                        if (rawRecord.IsNull() == false)
+                        {
+                            // can happen when we are in the process of restoring a database
+                            if (rawRecord.IsEncrypted() && MasterKey == null)
```

Comment:
you might read twice `IsEncrypted`. Move it into variable.

Reply:
d o n e